subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/ResolvableConfigurationsReportTaskIntegrationTest.groovy

```diff
@@ -590,6 +590,131 @@ The following Attributes have disambiguation rules defined.
         doesNotHaveIncubatingLegend()
     }
 
+    def "disambiguation rules are printed if added to attributes"() {
+        given: "A disambiguation rule applying to the alphabetically first named attribute in the list"
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            def flavor = Attribute.of('flavor', String)
+
+            configurations {
+                custom {
+                    description = "My custom configuration"
+                    canBeResolved = true
+                    canBeConsumed = false
+
+                    attributes {
+                        attribute(flavor, 'vanilla')
+                    }
+                }
+            }
+
+            class CategorySelectionRule implements AttributeDisambiguationRule<String> {
+                void execute(MultipleCandidatesDetails<String> details) {
+                    if (details.candidateValues.contains('chocolate')) {
+                        details.closestMatch('chocolate')
+                    }
+                }
+            }
+
+            dependencies.attributesSchema {
+                attribute(flavor) {
+                    disambiguationRules.add(CategorySelectionRule)
+                }
+
+                def usage = getAttributes().find { it.name == 'org.gradle.usage' }
+                setAttributeDisambiguationPrecedence([flavor, usage])
+            }
+        """.stripIndent()
+
+        when:
+        succeeds ':resolvableConfigurations'
+
+        then:
+        result.groupedOutput.task(":resolvableConfigurations").assertOutputContains("""--------------------------------------------------
+Disambiguation Rules
+--------------------------------------------------
+The following Attributes have disambiguation rules defined.
+
+    - flavor (1)
+    - org.gradle.category
+    - org.gradle.dependency.bundling
+    - org.gradle.jvm.environment
+    - org.gradle.jvm.version
+    - org.gradle.libraryelements
+    - org.gradle.plugin.api-version
+    - org.gradle.usage (2)
+
+(Attribute disambiguation precedence)""")
```

Comment:
I think this is unclear, it doesn't really tell me that `(n)` means the precedence. Maybe this works?:
```suggestion
(n): Attribute disambiguation precedence""")
```

Reply:
Or maybe `(#): Attribute disambiguation precedence`?

I don't feel strongly about this, but another way to do this would be to split and sort the attributes by this order followed by unordered attributes alphabetically.

Reply:
I don't think it's worth the screen space to list these twice since in most cases they are probably not something that would affect resolution (if there are exact matches, or there are not multiple compatible attributes, etc.).

I like the alternate message though, will update.